### DISCUSSION TITLE
Reaper shows the VST2 instances as having 64 channels and without appropriate naming

### DISCUSSION THREAD

**Discussion by dervur** [RomanKubiak/ctrlr#680] [Q&A]
Hello there,
Does anybody know why CtrlrX VST 2 and its exported instances appear as having 64 channels in Reaper? Is there a way to change this? The VST3 version seems to not be affected. Also the VST 2 exports do not have the given names of the panel in reaper and appear just with a CtrlrX name (similarly to what happens with VST3 that need to be changed in Projucer. 
![image](https://github.com/user-attachments/assets/098be97b-e95f-44ca-bce5-0197d122448c)
These have a build with the values changed in Projucer, also notice the 64ch
![image](https://github.com/user-attachments/assets/e6e86c92-b2a6-4e29-8dad-949139b7715b)

And this happens on when doing "Insert virtual instrument on new track..."
![image](https://github.com/user-attachments/assets/a336f152-dfe0-481f-9a79-9edba9ec5776)

Not a big problem but if there is a way to solve this it would be nice :)

**Comment by damiensellier** [comment]
The method used to name vst2 exports is different for DAWs and also OS dependent.
I can't tell for reaper unfortunately. 
Unless you can't use anything but vst2 I'd recommend using vst3 thoough.

We currently use the JUCE legacy mode for audio i/o mapping.
the i/o are set to stereo pairs : 
https://github.com/RomanKubiak/ctrlr/blob/8aa00d82127acda42ad9ac9b7b479461e9436aa4/Source/Plugin/CtrlrProcessor.cpp#L212C22-L212C46

It's a known reaper problem: 
https://forum.juce.com/t/reaper-plugin-stereo-audio-outputs-edit-solved/8865

I will update it in the processor.cpp file for the forthcoming update of ctrlrx to see if it changes something.

**Reply by dervur** [reply]
It must be just a reaper issue then, all good I was just curious tbh.
I'm not using VST3 because of the old issue I was having where there were weird MIDI messages in the VST3 exported instances, I will begin looking into that issue later on, and once it works I'll begin using VST3 as they appear to be more stable anyway.

**Comment by damiensellier** [comment]
You can replace your processor.cpp and processor.h files to see if it runs better with the "beta" ones.
I use it for my plugins and have no issue on my setup.
It goes in Source/Plugin/
[Source_Plugin.zip](https://github.com/user-attachments/files/16909308/Source_Plugin.zip)

**Reply by dervur** [reply]
I'll have a look at it over the weekend, thank you!

**Reply by damiensellier** [reply]
sure let me know if it works. now the mapping is set in the initial declaraction : 

```
CtrlrProcessor::CtrlrProcessor() :
                                    #ifndef JucePlugin_PreferredChannelConfigurations
                                    AudioProcessor (BusesProperties()
                                        #if ! JucePlugin_IsMidiEffect
                                        #if ! JucePlugin_IsSynth
                                                    .withInput  ("Input",  AudioChannelSet::stereo(), true) // stereo In
                                        #endif
                                                    .withOutput ("Output", AudioChannelSet::stereo(), true) // stereo Out
                                        #endif
                                                    ),
                                        #endif

                                    overridesTree (Ids::ctrlrOverrides),
                                    ctrlrManager (nullptr)
{
```

**Comment by dobo365** [comment]
In 5.3.201, the name of the VST2 is the one of the panel. You need to modify the VST ID manually to avoid duplicates. Some DAWs are relying on the VST name while others on the ID. If they rely on the ID then if the ID is duplicate you will only see one VST (the first one).
Reaper works fine and like the other DAWs. Nothing specific.

But interesting question, I didn't produce any VST2 with CtrlrX yet so will try that to see if it is the same.

And sorry, but I don't understand what you changed in the CtrlrProcessor files (what will it do?) but of course I'll replace them as well.

**Reply by damiensellier** [reply]
Hi Dominique, the modification is only the update to later versions of juce. The layout of audio i/o is more simple. You'll see in the comments I removed a lot of the older layout definitions, they were already legacy with JUCE 6. You'll tell me if it gives the proper stereo configuration on reaper.

Concerning vst2 names, I have a chart at home with different DAWs and WIN/macOS and how they pick up their plugin name in a different way. I remember that the vst2 passed **instigator | ctrlr** on windows with cubase but on cubase macOS it's just the filename of the plugin that identifies it. So a little bit confusing, nothing straight about this.

PS: je suis pas mal occupé en ce moment mais quand je suis libre je me remettrai à notre liste d'améliorations ;) Je t'avais fait une longue réponse à ton précédent email. Bonne rentrée!

**Reply by dervur** [reply]
I use the latest CtrlrX version 5.6.30
Yeah it appears each DAW does their own and each OS too. To be honest I am lately just compiling for each plugin as I would do for VST3, but since VST3 gives the weird response I already reported I use this for VST2 and it works fine. I guess this is the most safe way to do it in general (at least for Windows) even for VST2.
